EFCore.NamingConventions/Internal/NameRewritingConvention.cs

```diff
@@ -34,7 +34,7 @@ public NameRewritingConvention(INameRewriter nameRewriter)
 
         // Note that the base type is null when the entity type is first added - a base type only gets added later
         // (see ProcessEntityTypeBaseTypeChanged). But we still have this check for safety.
-        if (entityType.BaseType is null)
+        if (entityType.BaseType is null && !entityType.ClrType.IsAbstract)
```

Comment:
this obviously breaks that all abstract tables are not rewritten and would stay in a different case, when UseSnakeCaseNamingConvention is applied the tablename would stay in normal case, which would break a lot of existing code and tables.
related to #184